ocaml/typing/predef.ml

```diff
@@ -342,23 +342,35 @@ let build_initial_env add_type add_extension empty_env =
   |> add_type1 ident_iarray
        ~variance:Variance.covariant
        ~separability:Separability.Ind
+       ~param_jkind:(Jkind.Primitive.any_non_null ~why:Array_type_argument)
   |> add_type ident_bool
        ~kind:(variant [ cstr ident_false []; cstr ident_true []]
                 [| Constructor_uniform_value, [| |];
                    Constructor_uniform_value, [| |] |])
        ~jkind:(Jkind.Primitive.immediate ~why:Enumeration)
-  |> add_type ident_char ~jkind:(Jkind.Primitive.immediate ~why:(Primitive ident_char))
+       ~jkind_annotation:Jkind.Const.Primitive.immediate
+  |> add_type ident_char
+      ~jkind:(Jkind.Primitive.immediate ~why:(Primitive ident_char))
       ~jkind_annotation:Jkind.Const.Primitive.immediate
   |> add_type ident_exn
        ~kind:Type_open
        ~jkind:(Jkind.Primitive.value ~why:Extensible_variant)
   |> add_type ident_extension_constructor
   |> add_type ident_float
```

Comment:
This could also cross uniqueness. Though it's not hard to imagine allowing in-place updates of unique floating point numbers. Maybe better not to.

Reply:
Yes in-place update can in principle apply to any `immutable_data`.